gateway-server/src/main/java/org/apache/knox/gateway/GatewayServlet.java

```diff
@@ -277,4 +278,34 @@ public Enumeration<String> getInitParameterNames() {
       return config.getInitParameterNames();
     }
   }
+
+  private Exception sanitizeException(Exception e) {
+    if (e == null || e.getMessage() == null) {
+      return e;
+    }
+    if (!isErrorMessageSanitizationEnabled || e.getMessage() == null) {
+      return e;
+    }
+    String sanitizedMessage = e.getMessage().replaceAll("\\b(?:\\d{1,3}\\.){3}\\d{1,3}\\b", "[hidden]");
+    return createNewException(e, sanitizedMessage);
```

Comment:
I agree that a SanitizedException class to wrap other exceptions would be appropriate for this. The only reason for not doing this is if upstream catchers of these exceptions would ignore the SanitizedException because they're expecting what would be wrapped thereby.

Reply:
@moresandeep @pzampino I see the merit in having a dedicated class for sanitizing and logging, which can offer a more maintainable approach moving forward.

However, I have some concerns regarding potential breaking changes. Specifically, I'm not certain if there are users who might be utilizing `GatewayServlet` as if it were a library. If that's the case, changing the exception signature could lead to breaking changes for them. This is because, by introducing a new `SanitizedKnoxException`, we would be altering the types of exceptions that upstream code is expected to catch. Currently, throwing back the same exception type ensures backward compatibility and constitutes a non-breaking change.

Reply:
@moresandeep Logging the original error makes sense. Just committed the fix.

Reply:
Just implemented the SanitzedException.